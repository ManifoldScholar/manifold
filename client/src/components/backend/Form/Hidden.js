import React, { Component, PropTypes } from 'react';
import { Form } from 'components/backend';
import sharedPropsValidation from './propTypes';

const { label, ...validation } = sharedPropsValidation;

export default class FormHiddenInput extends Component {

  static displayName = "Form.HiddenInput";

  static propTypes = {
    ...validation
  };

  constructor(props) {
    super(props);
  }

  componentWillReceiveProps() {

  }

  render() {
    return (
      <Form.Connect.Set {...this.props} manualSet >
        <Form.Helpers.HiddenInput />
      </Form.Connect.Set>
    );
  }
}
