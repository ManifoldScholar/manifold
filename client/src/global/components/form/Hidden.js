import React, { Component } from "react";
import PropTypes from "prop-types";
import setter from "./setter";
import isNull from "lodash/isNull";

class FormHiddenInput extends Component {
  static displayName = "Form.HiddenInput";

  static propTypes = {
    value: PropTypes.any,
    onChange: PropTypes.func
  };

  render() {
    const value = isNull(this.props.value) ? "" : this.props.value;
    return (
      <div>
        <input type="hidden" value={value} onChange={this.props.onChange} />
      </div>
    );
  }
}

export default setter(FormHiddenInput);
