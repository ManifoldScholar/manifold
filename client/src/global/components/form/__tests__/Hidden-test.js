import Hidden from "../Hidden";

describe("global/components/form/Hidden", () => {
  def("mock", () => jest.fn());
  def("root", () => <Hidden value="attributes[fake]" onChange={$mock} />);
  it("matches the snapshot", () => {
    expect(shallow($root)).toMatchSnapshot();
  });
});
