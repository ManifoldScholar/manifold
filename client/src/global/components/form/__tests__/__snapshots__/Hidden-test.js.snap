// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global/components/form/Hidden matches the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
